prime-router/src/main/kotlin/fhirengine/translation/hl7/ConfigSchemaProcessor.kt

```diff
@@ -40,10 +40,8 @@ abstract class ConfigSchemaProcessor : Logging {
         }
 
         // when valueSet is available, use the matching value else just pass the value as is
-        // does a lowerCase comparison
-        if (retVal.isNotBlank() && element.valueSet.isNotEmpty()) {
-            val lowerSet = element.valueSet.mapKeys { it.key.lowercase() }
-            retVal = lowerSet.getOrDefault(retVal.lowercase(), retVal)
+        if (retVal.isNotBlank() && element.valueSet != null) {
+            retVal = element.valueSet?.getMappedValue(retVal) ?: retVal
```

Comment:
Why is this `?` required since we check for null?

Reply:
Since it's a `var` and not a `val`, it could theoretically (though nearly impossibly) be non-null when we check for it in L43, but then null when it is used in L44. So the compiler enforces this, we could either use `!!` to tell the compiler it's okay, or we could get rid of the check for null in L43. When we were pairing, we decided that it reads to match the comment in L42 more easily if we leave it like this, and using `?` instead of `!!` is still safer. On another look, I think we should just get rid of the null check in L43 and maybe reword the comment.

Reply:
Ah right.  I think in this particular case it could only be null if a line was added to set it to `null` so `!!` is safe, but yes I agree that just removing the check might make the most sense since it's redundant with the `?` operator